packages/core/src/v2/entity-deserializer.ts

```diff
@@ -79,13 +79,15 @@ function extractODataETag(json: MapType<any>): string | undefined {
 
 function getFieldValue<EntityT extends Entity, JsonT>(
   json: Partial<JsonT>,
-  selectable: Selectable<EntityT>
+  selectable: Field<EntityT> | Link<EntityT>
```

Comment:
I am confused now. The original selectable type contained:```  | SimpleTypeFields<EntityT>
  | Link<EntityT, any>
  | ComplexTypeField<EntityT>
  | CustomField<EntityT>
  | AllFields<EntityT>;
```
So quite a lot. In this PR this is reduced. Also from a logical point of view it seems like the implementation from getFieldValue  considers only EdmTypeField, Link and ComplexTypeField. So the type above should indicate this?

Reply:
This is related to your comment before. When checking this, I realized, that Selectable is not the correct type here. This function deserializes a field of an entity, which can either be a "property" or a "navigation property" (we call them "Field" and "Link" - my bad). What the function does is correct, but the type it took was too broad.